src/utils/daos_hdlr.c

```diff
@@ -1561,6 +1562,817 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
 	return rc;
 }
 
+static int
+parse_filename(const char* path, char** _obj_name, char** _cont_name)
+{
+	char *f1 = NULL;
+	char *f2 = NULL;
+	char *fname = NULL;
+	char *cont_name = NULL;
+	int rc = 0;
+	if (path == NULL || _obj_name == NULL || _cont_name == NULL)
+			return -EINVAL;
+
+	if (strcmp(path, "/") == 0) {
+		*_cont_name = strdup("/");
+		if (*_cont_name == NULL)
+			return -ENOMEM;
+		*_obj_name = NULL;
+		return 0;
+	}
+	f1 = strdup(path);
+	if (f1 == NULL) {
+		rc = -ENOMEM;
+		goto out;
+	}
+
+	f2 = strdup(path);
+	if (f2 == NULL) {
+		rc = -ENOMEM;
+		goto out;
+	}
+	fname = basename(f1);
+	cont_name = dirname(f2);
+
+	if (cont_name[0] != '/') {
+		char cwd[1024];
+
+		if (getcwd(cwd, 1024) == NULL) {
+			rc = -ENOMEM;
+			goto out;
+		}
+
+		if (strcmp(cont_name, ".") == 0) {
+			cont_name = strdup(cwd);
+			if (cont_name == NULL) {
+				rc = -ENOMEM;
+				goto out;
+			}
+		} else {
+			char *new_dir = calloc(strlen(cwd) + strlen(cont_name) + 1, sizeof(char));
+			if (new_dir == NULL) {
+				rc = -ENOMEM;
+				goto out;
+			}
+
+			strcpy(new_dir, cwd);
+			if (cont_name[0] == '.') {
+				strcat(new_dir, &cont_name[1]);
+			} else {
+				strcat(new_dir, "/");
+				strcat(new_dir, cont_name);
+			}
+			cont_name = new_dir;
+		}
+		*_cont_name = cont_name;
+	} else {
+		*_cont_name = strdup(cont_name);
+		if (*_cont_name == NULL) {
+			rc = -ENOMEM;
+			goto out;
+		}
+	}
+	*_obj_name = strdup(fname);
+	if (*_obj_name == NULL) {
+		free(*_cont_name);
+		*_cont_name = NULL;
+		rc = -ENOMEM;
+		goto out;
+	}
+	out:
+		if (f1)
+			free(f1);
+		if (f2)
+			free(f2);
+	return rc;
+}
+
+static ssize_t
+daos_write(struct daos_file_t *daos_file, char *file, void *buf, size_t size)
```

Comment:
daos_ prefix is for public API functions, so for internal functions we shouldn't be using it. (applies to all other static functions here)

Reply:
I think this ties into [DAOS-6098](https://jira.hpdd.intel.com/browse/DAOS-6098), where we need to decide where we want these functions to live, and how to name them.

Reply:
But I'm not sure what the name should be in the meantime.

Reply:
yea sure. if we are adding those to libdfs, it would be a dfs_ prefix. or maybe dfs_hlpr_ ..
but in this PR, since this is static function in this file only, it doesn't need a prefix. maybe just write_dfs_*

Reply:
@mchaarawi @daltonbohning added a _dfs to the end of the I/O functions and the file struct, instead of using the daos_ prefix